docs/build.sh

```diff
@@ -132,11 +132,6 @@ function generate_api_rst() {
 
 generate_api_rst v3
 
-# Fixup anchors and references in v3 so they form a distinct namespace.
-# TODO(htuch): Do this in protodoc generation in the future.
-find "${GENERATED_RST_DIR}"/api-v3 -name "*.rst" -print0 | xargs -0 sed -i -e "s#envoy_api_#envoy_v3_api_#g"
-find "${GENERATED_RST_DIR}"/api-v3 -name "*.rst" -print0 | xargs -0 sed -i -e "s#config_resource_monitors#v3_config_resource_monitors#g"
-
```

Comment:
this is the main change - ie not mangling the refs after the fact. They are kinda inconsistent in that in the proto files they sometimes do/not use the full `api_v3_etc` 

also the less mangling the better - eg the jsonschema wants to consume these descriptions too to turn them into markdown links

Reply:
the second of these is redundant now anyway